### ISSUE TITLE
Creating python environment for oggm-3dviz

### ISSUE THREAD

**Issue by bpelto** [OGGM/oggm-3dviz#11]
I tried to adapt my oggm environment to work with oggm-3d-viz, but after adding all required dependencies, I still get an error of:

ImportError: cannot import name 'vtk' from 'trame.widgets'

So I tried to create a separate environment using the yml here.

mamba env create -f environment.yml

But then get this error at the end:

Installing pip dependencies: | Ran pip subprocess with arguments:
['/home/bpelto/miniconda3/envs/oggm-3dviz/bin/python', '-m', 'pip', 'install', '-U', '-r', '/home/bpelto/oggm-3dviz/condaenv.hd8f66lr.requirements.txt', '--exists-action=b']
Pip subprocess output:

Pip subprocess error:
ERROR: Could not find a version that satisfies the requirement xpystack (from versions: none)
ERROR: No matching distribution found for xpystack

failed

CondaEnvException: Pip failed

Any tips? Thanks!

**Comment by benbovy** [comment]
> ERROR: No matching distribution found for xpystack

Not sure what causes this error but with #12 this package won't be needed anymore.

**Comment by pat-schmitt** [comment]
Yes, for me the creation of the environment worked with the updates from https://github.com/OGGM/oggm-3dviz/pull/12

**Comment by bpelto** [comment]
Thanks @benbovy & @pat-schmitt , I started fresh with the updates from #12 and this worked for me too. 

Then when testing the getting_started.ipynb, I can't get the viz.show() to work (running in VScode on WSL: Ubuntu).

Perhaps I need to install xsfb?

error:
/home/bpelto/miniconda3/envs/oggm-3dviz/lib/python3.12/site-packages/pyvista/plotting/plotter.py:151: UserWarning: 
This system does not appear to be running an xserver.
PyVista will likely segfault when rendering.

Try starting a virtual frame buffer with xvfb, or using
  ``pyvista.start_xvfb()`
  
  ....the Kernel crashed while executing ....
  
  My install of oggm-3dviz is:
  	mamba create env -f environment.yml
	mamba install pip
        pip install -e .  (install with pip from oggm-3dviz cloned repo)  
        mamba install -c conda-forge ipywidgets             (just in case noting your link in the notebook about having ipywidgets correctly installed -- but was already satisfied)

Thanks for any insights!

**Comment by benbovy** [comment]
This seems to be related to pyvista and/or its rendering backend (trame) rather than an issue with oggm-3dviz. I don't know if pyvista/trame supports VSCode. I don't use VSCode but I know that it has its own implementation of the jupyter widget protocol, which might causes some issues as well. The notebooks in the tutorial should work fine in Jupyterlab or Jupyter Notebook front-ends.

**Comment by bpelto** [comment]
Thanks @benbovy, I tried running in jupyter lab but got the same error, then found this info on Pyvista on Running on WSL: https://docs.pyvista.org/version/stable/getting-started/installation.html#running-on-wsl

Which did the trick for me!
